File: client/ruby/lib/swagger_client/api/typeclasses_api.rb
require "uri"

module SwaggerClient
  class TypeclassesApi
    attr_accessor :api_client

    def initialize(api_client = nil)
      @api_client = api_client || Configuration.api_client
    end

    # List
    # 
    # @param [Hash] opts the optional parameters
    # @option opts [Integer] :offset 
    # @option opts [Integer] :limit 
    # @return [Array<Typedescription>]
    def list(opts = {})
      if Configuration.debugging
        Configuration.logger.debug "Calling API: TypeclassesApi#list ..."
      end
      
      # resource path
      path = "/typeclasses".sub('{format}','json')

      # query parameters
      query_params = {}
      query_params[:'offset'] = opts[:'offset'] if opts[:'offset']
      query_params[:'limit'] = opts[:'limit'] if opts[:'limit']

      # header parameters
      header_params = {}

      # HTTP header 'Accept' (if needed)
      _header_accept = []
      _header_accept_result = @api_client.select_header_accept(_header_accept) and header_params['Accept'] = _header_accept_result

      # HTTP header 'Content-Type'
      _header_content_type = []
      header_params['Content-Type'] = @api_client.select_header_content_type(_header_content_type)

      # form parameters
      form_params = {}

      # http body (model)
      post_body = nil
      

      auth_names = []
      result = @api_client.call_api(:GET, path,
        :header_params => header_params,
        :query_params => query_params,
        :form_params => form_params,
        :body => post_body,
        :auth_names => auth_names,
        :return_type => 'Array<Typedescription>')
      if Configuration.debugging
        Configuration.logger.debug "API called: TypeclassesApi#list. Result: #{result.inspect}"
      end
      return result
    end

    # Create
    # 
    # @param [Hash] opts the optional parameters
    # @option opts [Typedescription] :body 
    # @return [Typedescription]
    def create(opts = {})
      if Configuration.debugging
        Configuration.logger.debug "Calling API: TypeclassesApi#create ..."
      end
      
      # resource path
      path = "/typeclasses".sub('{format}','json')

      # query parameters
      query_params = {}

      # header parameters
      header_params = {}

      # HTTP header 'Accept' (if needed)
      _header_accept = []
      _header_accept_result = @api_client.select_header_accept(_header_accept) and header_params['Accept'] = _header_accept_result

      # HTTP header 'Content-Type'
      _header_content_type = []
      header_params['Content-Type'] = @api_client.select_header_content_type(_header_content_type)

      # form parameters
      form_params = {}

      # http body (model)
      post_body = @api_client.object_to_http_body(opts[:'body'])
      

      auth_names = []
      result = @api_client.call_api(:POST, path,
        :header_params => header_params,
        :query_params => query_params,
        :form_params => form_params,
        :body => post_body,
        :auth_names => auth_names,
        :return_type => 'Typedescription')
      if Configuration.debugging
        Configuration.logger.debug "API called: TypeclassesApi#create. Result: #{result.inspect}"
      end
      return result
    end

    # Read
    # 
    # @param id 
    # @param [Hash] opts the optional parameters
    # @return [Typedescription]
    def read(id, opts = {})
      if Configuration.debugging
        Configuration.logger.debug "Calling API: TypeclassesApi#read ..."
      end
      
      # verify the required parameter 'id' is set
      fail "Missing the required parameter 'id' when calling read" if id.nil?
      
      # resource path
      path = "/typeclasses/{id}".sub('{format}','json').sub('{' + 'id' + '}', id.to_s)

      # query parameters
      query_params = {}

      # header parameters
      header_params = {}

      # HTTP header 'Accept' (if needed)
      _header_accept = []
      _header_accept_result = @api_client.select_header_accept(_header_accept) and header_params['Accept'] = _header_accept_result

      # HTTP header 'Content-Type'
      _header_content_type = []
      header_params['Content-Type'] = @api_client.select_header_content_type(_header_content_type)

      # form parameters
      form_params = {}

      # http body (model)
      post_body = nil
      

      auth_names = []
      result = @api_client.call_api(:GET, path,
        :header_params => header_params,
        :query_params => query_params,
        :form_params => form_params,
        :body => post_body,
        :auth_names => auth_names,
        :return_type => 'Typedescription')
      if Configuration.debugging
        Configuration.logger.debug "API called: TypeclassesApi#read. Result: #{result.inspect}"
      end
      return result
    end

    # Update
    # 
    # @param [Hash] opts the optional parameters
    # @option opts [Typedescription] :body 
    # @return [Typedescription]
    def update(opts = {})
      if Configuration.debugging
        Configuration.logger.debug "Calling API: TypeclassesApi#update ..."
      end
      
      # resource path
      path = "/typeclasses/{id}".sub('{format}','json')

      # query parameters
      query_params = {}

      # header parameters
      header_params = {}

      # HTTP header 'Accept' (if needed)
      _header_accept = []
      _header_accept_result = @api_client.select_header_accept(_header_accept) and header_params['Accept'] = _header_accept_result

      # HTTP header 'Content-Type'
      _header_content_type = []
      header_params['Content-Type'] = @api_client.select_header_content_type(_header_content_type)

      # form parameters
      form_params = {}

      # http body (model)
      post_body = @api_client.object_to_http_body(opts[:'body'])
      

      auth_names = []
      result = @api_client.call_api(:PUT, path,
        :header_params => header_params,
        :query_params => query_params,
        :form_params => form_params,
        :body => post_body,
        :auth_names => auth_names,
        :return_type => 'Typedescription')
      if Configuration.debugging
        Configuration.logger.debug "API called: TypeclassesApi#update. Result: #{result.inspect}"
      end
      return result
    end

    # Delete
    # 
    # @param id 
    # @param [Hash] opts the optional parameters
    # @return [nil]
    def delete(id, opts = {})
      if Configuration.debugging
        Configuration.logger.debug "Calling API: TypeclassesApi#delete ..."
      end
      
      # verify the required parameter 'id' is set
      fail "Missing the required parameter 'id' when calling delete" if id.nil?
      
      # resource path
      path = "/typeclasses/{id}".sub('{format}','json').sub('{' + 'id' + '}', id.to_s)

      # query parameters
      query_params = {}

      # header parameters
      header_params = {}

      # HTTP header 'Accept' (if needed)
      _header_accept = []
      _header_accept_result = @api_client.select_header_accept(_header_accept) and header_params['Accept'] = _header_accept_result

      # HTTP header 'Content-Type'
      _header_content_type = []
      header_params['Content-Type'] = @api_client.select_header_content_type(_header_content_type)

      # form parameters
      form_params = {}

      # http body (model)
      post_body = nil
      

      auth_names = []
      @api_client.call_api(:DELETE, path,
        :header_params => header_params,
        :query_params => query_params,
        :form_params => form_params,
        :body => post_body,
        :auth_names => auth_names)
      if Configuration.debugging
        Configuration.logger.debug "API called: TypeclassesApi#delete"
      end
      return nil
    end

    # List global types
    # 
    # @param typeclassid 
    # @param [Hash] opts the optional parameters
    # @return [nil]
    def get_global_type_for_type_class(typeclassid, opts = {})
      if Configuration.debugging
        Configuration.logger.debug "Calling API: TypeclassesApi#get_global_type_for_type_class ..."
      end
      
      # verify the required parameter 'typeclassid' is set
      fail "Missing the required parameter 'typeclassid' when calling get_global_type_for_type_class" if typeclassid.nil?
      
      # resource path
      path = "/typeclasses/{typeclassid}/typevalues".sub('{format}','json').sub('{' + 'typeclassid' + '}', typeclassid.to_s)

      # query parameters
      query_params = {}

      # header parameters
      header_params = {}

      # HTTP header 'Accept' (if needed)
      _header_accept = []
      _header_accept_result = @api_client.select_header_accept(_header_accept) and header_params['Accept'] = _header_accept_result

      # HTTP header 'Content-Type'
      _header_content_type = []
      header_params['Content-Type'] = @api_client.select_header_content_type(_header_content_type)

      # form parameters
      form_params = {}

      # http body (model)
      post_body = nil
      

      auth_names = []
      @api_client.call_api(:GET, path,
        :header_params => header_params,
        :query_params => query_params,
        :form_params => form_params,
        :body => post_body,
        :auth_names => auth_names)
      if Configuration.debugging
        Configuration.logger.debug "API called: TypeclassesApi#get_global_type_for_type_class"
      end
      return nil
    end

    # Create global type
    # 
    # @param typeclassid 
    # @param [Hash] opts the optional parameters
    # @option opts [Globaltype] :body 
    # @return [Globaltype]
    def create_global_type(typeclassid, opts = {})
      if Configuration.debugging
        Configuration.logger.debug "Calling API: TypeclassesApi#create_global_type ..."
      end
      
      # verify the required parameter 'typeclassid' is set
      fail "Missing the required parameter 'typeclassid' when calling create_global_type" if typeclassid.nil?
      
      # resource path
      path = "/typeclasses/{typeclassid}/typevalues".sub('{format}','json').sub('{' + 'typeclassid' + '}', typeclassid.to_s)

      # query parameters
      query_params = {}

      # header parameters
      header_params = {}

      # HTTP header 'Accept' (if needed)
      _header_accept = []
      _header_accept_result = @api_client.select_header_accept(_header_accept) and header_params['Accept'] = _header_accept_result

      # HTTP header 'Content-Type'
      _header_content_type = []
      header_params['Content-Type'] = @api_client.select_header_content_type(_header_content_type)

      # form parameters
      form_params = {}

      # http body (model)
      post_body = @api_client.object_to_http_body(opts[:'body'])
      

      auth_names = []
      result = @api_client.call_api(:POST, path,
        :header_params => header_params,
        :query_params => query_params,
        :form_params => form_params,
        :body => post_body,
        :auth_names => auth_names,
        :return_type => 'Globaltype')
      if Configuration.debugging
        Configuration.logger.debug "API called: TypeclassesApi#create_global_type. Result: #{result.inspect}"
      end
      return result
    end

    # Read global type
    # 
    # @param typeclassid 
    # @param typevalueid 
    # @param [Hash] opts the optional parameters
    # @return [Globaltype]
    def get_global_type(typeclassid, typevalueid, opts = {})
      if Configuration.debugging
        Configuration.logger.debug "Calling API: TypeclassesApi#get_global_type ..."
      end
      
      # verify the required parameter 'typeclassid' is set
      fail "Missing the required parameter 'typeclassid' when calling get_global_type" if typeclassid.nil?
      
      # verify the required parameter 'typevalueid' is set
      fail "Missing the required parameter 'typevalueid' when calling get_global_type" if typevalueid.nil?
      
      # resource path
      path = "/typeclasses/{typeclassid}/typevalues/{typevalueid}".sub('{format}','json').sub('{' + 'typeclassid' + '}', typeclassid.to_s).sub('{' + 'typevalueid' + '}', typevalueid.to_s)

      # query parameters
      query_params = {}

      # header parameters
      header_params = {}

      # HTTP header 'Accept' (if needed)
      _header_accept = []
      _header_accept_result = @api_client.select_header_accept(_header_accept) and header_params['Accept'] = _header_accept_result

      # HTTP header 'Content-Type'
      _header_content_type = []
      header_params['Content-Type'] = @api_client.select_header_content_type(_header_content_type)

      # form parameters
      form_params = {}

      # http body (model)
      post_body = nil
      

      auth_names = []
      result = @api_client.call_api(:GET, path,
        :header_params => header_params,
        :query_params => query_params,
        :form_params => form_params,
        :body => post_body,
        :auth_names => auth_names,
        :return_type => 'Globaltype')
      if Configuration.debugging
        Configuration.logger.debug "API called: TypeclassesApi#get_global_type. Result: #{result.inspect}"
      end
      return result
    end

    # Update global type
    # 
    # @param typeclassid 
    # @param typevalueid 
    # @param [Hash] opts the optional parameters
    # @option opts [Globaltype] :body 
    # @return [Globaltype]
    def update_global_type(typeclassid, typevalueid, opts = {})
      if Configuration.debugging
        Configuration.logger.debug "Calling API: TypeclassesApi#update_global_type ..."
      end
      
      # verify the required parameter 'typeclassid' is set
      fail "Missing the required parameter 'typeclassid' when calling update_global_type" if typeclassid.nil?
      
      # verify the required parameter 'typevalueid' is set
      fail "Missing the required parameter 'typevalueid' when calling update_global_type" if typevalueid.nil?
      
      # resource path
      path = "/typeclasses/{typeclassid}/typevalues/{typevalueid}".sub('{format}','json').sub('{' + 'typeclassid' + '}', typeclassid.to_s).sub('{' + 'typevalueid' + '}', typevalueid.to_s)

      # query parameters
      query_params = {}

      # header parameters
      header_params = {}

      # HTTP header 'Accept' (if needed)
      _header_accept = []
      _header_accept_result = @api_client.select_header_accept(_header_accept) and header_params['Accept'] = _header_accept_result

      # HTTP header 'Content-Type'
      _header_content_type = []
      header_params['Content-Type'] = @api_client.select_header_content_type(_header_content_type)

      # form parameters
      form_params = {}

      # http body (model)
      post_body = @api_client.object_to_http_body(opts[:'body'])
      

      auth_names = []
      result = @api_client.call_api(:PUT, path,
        :header_params => header_params,
        :query_params => query_params,
        :form_params => form_params,
        :body => post_body,
        :auth_names => auth_names,
        :return_type => 'Globaltype')
      if Configuration.debugging
        Configuration.logger.debug "API called: TypeclassesApi#update_global_type. Result: #{result.inspect}"
      end
      return result
    end

    # Delete global type
    # 
    # @param typeclassid 
    # @param typevalueid 
    # @param [Hash] opts the optional parameters
    # @return [nil]
    def delete_global_type(typeclassid, typevalueid, opts = {})
      if Configuration.debugging
        Configuration.logger.debug "Calling API: TypeclassesApi#delete_global_type ..."
      end
      
      # verify the required parameter 'typeclassid' is set
      fail "Missing the required parameter 'typeclassid' when calling delete_global_type" if typeclassid.nil?
      
      # verify the required parameter 'typevalueid' is set
      fail "Missing the required parameter 'typevalueid' when calling delete_global_type" if typevalueid.nil?
      
      # resource path
      path = "/typeclasses/{typeclassid}/typevalues/{typevalueid}".sub('{format}','json').sub('{' + 'typeclassid' + '}', typeclassid.to_s).sub('{' + 'typevalueid' + '}', typevalueid.to_s)

      # query parameters
      query_params = {}

      # header parameters
      header_params = {}

      # HTTP header 'Accept' (if needed)
      _header_accept = []
      _header_accept_result = @api_client.select_header_accept(_header_accept) and header_params['Accept'] = _header_accept_result

      # HTTP header 'Content-Type'
      _header_content_type = []
      header_params['Content-Type'] = @api_client.select_header_content_type(_header_content_type)

      # form parameters
      form_params = {}

      # http body (model)
      post_body = nil
      

      auth_names = []
      @api_client.call_api(:DELETE, path,
        :header_params => header_params,
        :query_params => query_params,
        :form_params => form_params,
        :body => post_body,
        :auth_names => auth_names)
      if Configuration.debugging
        Configuration.logger.debug "API called: TypeclassesApi#delete_global_type"
      end
      return nil
    end
  end
end

File: client/java/src/main/java/io/swagger/client/api/TypeclassesApi.java
package io.swagger.client.api;

import io.swagger.client.ApiException;
import io.swagger.client.ApiClient;
import io.swagger.client.Configuration;
import io.swagger.client.Pair;
import io.swagger.client.TypeRef;

import io.swagger.client.model.Typedescription;
import io.swagger.client.model.Globaltype;

import java.util.*;

@javax.annotation.Generated(value = "class io.swagger.codegen.languages.JavaClientCodegen", date = "2015-10-20T11:32:47.715-07:00")
public class TypeclassesApi {
  private ApiClient apiClient;

  public TypeclassesApi() {
    this(Configuration.getDefaultApiClient());
  }

  public TypeclassesApi(ApiClient apiClient) {
    this.apiClient = apiClient;
  }

  public ApiClient getApiClient() {
    return apiClient;
  }

  public void setApiClient(ApiClient apiClient) {
    this.apiClient = apiClient;
  }

  
  /**
   * List
   * 
   * @param offset 
   * @param limit 
   * @return List<Typedescription>
   */
  public List<Typedescription> list (Integer offset, Integer limit) throws ApiException {
    Object postBody = null;
    byte[] postBinaryBody = null;
    
    // create path and map variables
    String path = "/typeclasses".replaceAll("\\{format\\}","json");

    // query params
    List<Pair> queryParams = new ArrayList<Pair>();
    Map<String, String> headerParams = new HashMap<String, String>();
    Map<String, Object> formParams = new HashMap<String, Object>();

    
    queryParams.addAll(apiClient.parameterToPairs("", "offset", offset));
    
    queryParams.addAll(apiClient.parameterToPairs("", "limit", limit));
    

    

    

    final String[] accepts = {
      
    };
    final String accept = apiClient.selectHeaderAccept(accepts);

    final String[] contentTypes = {
      
    };
    final String contentType = apiClient.selectHeaderContentType(contentTypes);

    String[] authNames = new String[] {  };

    

    
    
    TypeRef returnType = new TypeRef<List<Typedescription>>() {};
    return apiClient.invokeAPI(path, "GET", queryParams, postBody, postBinaryBody, headerParams, formParams, accept, contentType, authNames, returnType);
    
    


  }
  
  /**
   * Create
   * 
   * @param body 
   * @return Typedescription
   */
  public Typedescription create (Typedescription body) throws ApiException {
    Object postBody = body;
    byte[] postBinaryBody = null;
    
    // create path and map variables
    String path = "/typeclasses".replaceAll("\\{format\\}","json");

    // query params
    List<Pair> queryParams = new ArrayList<Pair>();
    Map<String, String> headerParams = new HashMap<String, String>();
    Map<String, Object> formParams = new HashMap<String, Object>();

    

    

    

    final String[] accepts = {
      
    };
    final String accept = apiClient.selectHeaderAccept(accepts);

    final String[] contentTypes = {
      
    };
    final String contentType = apiClient.selectHeaderContentType(contentTypes);

    String[] authNames = new String[] {  };

    

    
    
    TypeRef returnType = new TypeRef<Typedescription>() {};
    return apiClient.invokeAPI(path, "POST", queryParams, postBody, postBinaryBody, headerParams, formParams, accept, contentType, authNames, returnType);
    
    


  }
  
  /**
   * Read
   * 
   * @param id 
   * @return Typedescription
   */
  public Typedescription read (Integer id) throws ApiException {
    Object postBody = null;
    byte[] postBinaryBody = null;
    
     // verify the required parameter 'id' is set
     if (id == null) {
        throw new ApiException(400, "Missing the required parameter 'id' when calling read");
     }
     
    // create path and map variables
    String path = "/typeclasses/{id}".replaceAll("\\{format\\}","json")
      .replaceAll("\\{" + "id" + "\\}", apiClient.escapeString(id.toString()));

    // query params
    List<Pair> queryParams = new ArrayList<Pair>();
    Map<String, String> headerParams = new HashMap<String, String>();
    Map<String, Object> formParams = new HashMap<String, Object>();

    

    

    

    final String[] accepts = {
      
    };
    final String accept = apiClient.selectHeaderAccept(accepts);

    final String[] contentTypes = {
      
    };
    final String contentType = apiClient.selectHeaderContentType(contentTypes);

    String[] authNames = new String[] {  };

    

    
    
    TypeRef returnType = new TypeRef<Typedescription>() {};
    return apiClient.invokeAPI(path, "GET", queryParams, postBody, postBinaryBody, headerParams, formParams, accept, contentType, authNames, returnType);
    
    


  }
  
  /**
   * Update
   * 
   * @param body 
   * @return Typedescription
   */
  public Typedescription update (Typedescription body) throws ApiException {
    Object postBody = body;
    byte[] postBinaryBody = null;
    
    // create path and map variables
    String path = "/typeclasses/{id}".replaceAll("\\{format\\}","json");

    // query params
    List<Pair> queryParams = new ArrayList<Pair>();
    Map<String, String> headerParams = new HashMap<String, String>();
    Map<String, Object> formParams = new HashMap<String, Object>();

    

    

    

    final String[] accepts = {
      
    };
    final String accept = apiClient.selectHeaderAccept(accepts);

    final String[] contentTypes = {
      
    };
    final String contentType = apiClient.selectHeaderContentType(contentTypes);

    String[] authNames = new String[] {  };

    

    
    
    TypeRef returnType = new TypeRef<Typedescription>() {};
    return apiClient.invokeAPI(path, "PUT", queryParams, postBody, postBinaryBody, headerParams, formParams, accept, contentType, authNames, returnType);
    
    


  }
  
  /**
   * Delete
   * 
   * @param id 
   * @return void
   */
  public void delete (Integer id) throws ApiException {
    Object postBody = null;
    byte[] postBinaryBody = null;
    
     // verify the required parameter 'id' is set
     if (id == null) {
        throw new ApiException(400, "Missing the required parameter 'id' when calling delete");
     }
     
    // create path and map variables
    String path = "/typeclasses/{id}".replaceAll("\\{format\\}","json")
      .replaceAll("\\{" + "id" + "\\}", apiClient.escapeString(id.toString()));

    // query params
    List<Pair> queryParams = new ArrayList<Pair>();
    Map<String, String> headerParams = new HashMap<String, String>();
    Map<String, Object> formParams = new HashMap<String, Object>();

    

    

    

    final String[] accepts = {
      
    };
    final String accept = apiClient.selectHeaderAccept(accepts);

    final String[] contentTypes = {
      
    };
    final String contentType = apiClient.selectHeaderContentType(contentTypes);

    String[] authNames = new String[] {  };

    

    
    
    apiClient.invokeAPI(path, "DELETE", queryParams, postBody, postBinaryBody, headerParams, formParams, accept, contentType, authNames, null);
    
    


  }
  
  /**
   * List global types
   * 
   * @param typeclassid 
   * @return void
   */
  public void getGlobalTypeForTypeClass (Integer typeclassid) throws ApiException {
    Object postBody = null;
    byte[] postBinaryBody = null;
    
     // verify the required parameter 'typeclassid' is set
     if (typeclassid == null) {
        throw new ApiException(400, "Missing the required parameter 'typeclassid' when calling getGlobalTypeForTypeClass");
     }
     
    // create path and map variables
    String path = "/typeclasses/{typeclassid}/typevalues".replaceAll("\\{format\\}","json")
      .replaceAll("\\{" + "typeclassid" + "\\}", apiClient.escapeString(typeclassid.toString()));

    // query params
    List<Pair> queryParams = new ArrayList<Pair>();
    Map<String, String> headerParams = new HashMap<String, String>();
    Map<String, Object> formParams = new HashMap<String, Object>();

    

    

    

    final String[] accepts = {
      
    };
    final String accept = apiClient.selectHeaderAccept(accepts);

    final String[] contentTypes = {
      
    };
    final String contentType = apiClient.selectHeaderContentType(contentTypes);

    String[] authNames = new String[] {  };

    

    
    
    apiClient.invokeAPI(path, "GET", queryParams, postBody, postBinaryBody, headerParams, formParams, accept, contentType, authNames, null);
    
    


  }
  
  /**
   * Create global type
   * 
   * @param typeclassid 
   * @param body 
   * @return Globaltype
   */
  public Globaltype createGlobalType (Integer typeclassid, Globaltype body) throws ApiException {
    Object postBody = body;
    byte[] postBinaryBody = null;
    
     // verify the required parameter 'typeclassid' is set
     if (typeclassid == null) {
        throw new ApiException(400, "Missing the required parameter 'typeclassid' when calling createGlobalType");
     }
     
    // create path and map variables
    String path = "/typeclasses/{typeclassid}/typevalues".replaceAll("\\{format\\}","json")
      .replaceAll("\\{" + "typeclassid" + "\\}", apiClient.escapeString(typeclassid.toString()));

    // query params
    List<Pair> queryParams = new ArrayList<Pair>();
    Map<String, String> headerParams = new HashMap<String, String>();
    Map<String, Object> formParams = new HashMap<String, Object>();

    

    

    

    final String[] accepts = {
      
    };
    final String accept = apiClient.selectHeaderAccept(accepts);

    final String[] contentTypes = {
      
    };
    final String contentType = apiClient.selectHeaderContentType(contentTypes);

    String[] authNames = new String[] {  };

    

    
    
    TypeRef returnType = new TypeRef<Globaltype>() {};
    return apiClient.invokeAPI(path, "POST", queryParams, postBody, postBinaryBody, headerParams, formParams, accept, contentType, authNames, returnType);
    
    


  }
  
  /**
   * Read global type
   * 
   * @param typeclassid 
   * @param typevalueid 
   * @return Globaltype
   */
  public Globaltype getGlobalType (Integer typeclassid, Integer typevalueid) throws ApiException {
    Object postBody = null;
    byte[] postBinaryBody = null;
    
     // verify the required parameter 'typeclassid' is set
     if (typeclassid == null) {
        throw new ApiException(400, "Missing the required parameter 'typeclassid' when calling getGlobalType");
     }
     
     // verify the required parameter 'typevalueid' is set
     if (typevalueid == null) {
        throw new ApiException(400, "Missing the required parameter 'typevalueid' when calling getGlobalType");
     }
     
    // create path and map variables
    String path = "/typeclasses/{typeclassid}/typevalues/{typevalueid}".replaceAll("\\{format\\}","json")
      .replaceAll("\\{" + "typeclassid" + "\\}", apiClient.escapeString(typeclassid.toString()))
      .replaceAll("\\{" + "typevalueid" + "\\}", apiClient.escapeString(typevalueid.toString()));

    // query params
    List<Pair> queryParams = new ArrayList<Pair>();
    Map<String, String> headerParams = new HashMap<String, String>();
    Map<String, Object> formParams = new HashMap<String, Object>();

    

    

    

    final String[] accepts = {
      
    };
    final String accept = apiClient.selectHeaderAccept(accepts);

    final String[] contentTypes = {
      
    };
    final String contentType = apiClient.selectHeaderContentType(contentTypes);

    String[] authNames = new String[] {  };

    

    
    
    TypeRef returnType = new TypeRef<Globaltype>() {};
    return apiClient.invokeAPI(path, "GET", queryParams, postBody, postBinaryBody, headerParams, formParams, accept, contentType, authNames, returnType);
    
    


  }
  
  /**
   * Update global type
   * 
   * @param typeclassid 
   * @param typevalueid 
   * @param body 
   * @return Globaltype
   */
  public Globaltype updateGlobalType (Integer typeclassid, Integer typevalueid, Globaltype body) throws ApiException {
    Object postBody = body;
    byte[] postBinaryBody = null;
    
     // verify the required parameter 'typeclassid' is set
     if (typeclassid == null) {
        throw new ApiException(400, "Missing the required parameter 'typeclassid' when calling updateGlobalType");
     }
     
     // verify the required parameter 'typevalueid' is set
     if (typevalueid == null) {
        throw new ApiException(400, "Missing the required parameter 'typevalueid' when calling updateGlobalType");
     }
     
    // create path and map variables
    String path = "/typeclasses/{typeclassid}/typevalues/{typevalueid}".replaceAll("\\{format\\}","json")
      .replaceAll("\\{" + "typeclassid" + "\\}", apiClient.escapeString(typeclassid.toString()))
      .replaceAll("\\{" + "typevalueid" + "\\}", apiClient.escapeString(typevalueid.toString()));

    // query params
    List<Pair> queryParams = new ArrayList<Pair>();
    Map<String, String> headerParams = new HashMap<String, String>();
    Map<String, Object> formParams = new HashMap<String, Object>();

    

    

    

    final String[] accepts = {
      
    };
    final String accept = apiClient.selectHeaderAccept(accepts);

    final String[] contentTypes = {
      
    };
    final String contentType = apiClient.selectHeaderContentType(contentTypes);

    String[] authNames = new String[] {  };

    

    
    
    TypeRef returnType = new TypeRef<Globaltype>() {};
    return apiClient.invokeAPI(path, "PUT", queryParams, postBody, postBinaryBody, headerParams, formParams, accept, contentType, authNames, returnType);
    
    


  }
  
  /**
   * Delete global type
   * 
   * @param typeclassid 
   * @param typevalueid 
   * @return void
   */
  public void deleteGlobalType (Integer typeclassid, Integer typevalueid) throws ApiException {
    Object postBody = null;
    byte[] postBinaryBody = null;
    
     // verify the required parameter 'typeclassid' is set
     if (typeclassid == null) {
        throw new ApiException(400, "Missing the required parameter 'typeclassid' when calling deleteGlobalType");
     }
     
     // verify the required parameter 'typevalueid' is set
     if (typevalueid == null) {
        throw new ApiException(400, "Missing the required parameter 'typevalueid' when calling deleteGlobalType");
     }
     
    // create path and map variables
    String path = "/typeclasses/{typeclassid}/typevalues/{typevalueid}".replaceAll("\\{format\\}","json")
      .replaceAll("\\{" + "typeclassid" + "\\}", apiClient.escapeString(typeclassid.toString()))
      .replaceAll("\\{" + "typevalueid" + "\\}", apiClient.escapeString(typevalueid.toString()));

    // query params
    List<Pair> queryParams = new ArrayList<Pair>();
    Map<String, String> headerParams = new HashMap<String, String>();
    Map<String, Object> formParams = new HashMap<String, Object>();

    

    

    

    final String[] accepts = {
      
    };
    final String accept = apiClient.selectHeaderAccept(accepts);

    final String[] contentTypes = {
      
    };
    final String contentType = apiClient.selectHeaderContentType(contentTypes);

    String[] authNames = new String[] {  };

    

    
    
    apiClient.invokeAPI(path, "DELETE", queryParams, postBody, postBinaryBody, headerParams, formParams, accept, contentType, authNames, null);
    
    


  }
  
}
